evap/evaluation/templates/base.html

```diff
@@ -85,41 +83,75 @@
             };
-            applySelect2($('select').not('.form-template select').not('[disabled]').not('.no-select2').not('#id_delegate_to'));
-            // handle delegation modal, which includes a dropdown select field
-            applySelect2($('select#id_delegate_to'), $('#delegateSelectionModal'));
 
-            // don't use the placeholder text on disabled select2 elements
-            applySelect2Disabled = function(elem) {
-                elem.select2();
+            class MinimumInputLengthTomSelect extends TomSelect {
+                search(query) {
+                    if(query.length < this.settings.minimumInputLength) {
+                        return this.sifter.prepareSearch(query, {});
+                    }
+                    return super.search(query);
+                }
+            }
+
+            applyTomSelect = function(elements, additionalOptions = {}) {
+                elements.forEach(function(element) {
+                    const minimumInputLength = element.options.length >= 50 ? 3 : 0;
+
+                    if(element.tomselect) {
+                        element.tomselect.destroy();
+                    }
+                    element.classList.remove("form-select");  // TomSelect applies their own matching classes / styles
+                    if(element.parentElement.classList.contains("is-invalid")) {  // hotfix for our wrongly placed is-invalid
+                        element.parentElement.classList.remove("is-invalid");
+                        element.classList.add("is-invalid");
+                    }
+                    const baseOptions = {
+                        createOnBlur: true,
+                        placeholder: "{% trans 'Please select...' %}",
+                        hidePlaceholder: true,
+                        minimumInputLength: minimumInputLength,
+                        render: {
+                            option_create: function( data, escape ) {return '<div class="create">' + escape(data.input) + '</div>';},
+                            no_results: function( data, escape ) {
+                                if(data.input.length < minimumInputLength) {
+                                    return `<div class="no-results">{% trans "Please enter ${minimumInputLength} characters or more..." %}</div>`;
+                                } else {
+                                    return '<div class="no-results">{% trans "No results found" %}</div>';
+                                }
+                            }
+                        },
+                        plugins: {},
+                    };
+                    if(element.hasAttribute("data-tomselect-fullwidth")) {
+                        // span needed to the "remove this icon" button is right-aligned
+                        baseOptions.render.item = function(data, escape) {return '<div class="w-100"><span class="w-100">' + escape(data.text) + '</span></div>';};
```

Comment:
This (and similar online functions) could use the arrow function syntax. Additionally, this could use template strings, not sure which would be more readable though